drivers/bluetooth/hci/hci_stm32wba.c

```diff
@@ -0,0 +1,380 @@
+		return NULL;
+	}
+
+	discardable = is_hci_event_discardable(data);
+
+	memcpy((void *)&hdr, data, sizeof(hdr));
+	data += sizeof(hdr);
+	len -= sizeof(hdr);
+
+	if (len != hdr.len) {
+		LOG_ERR("Event payload length is not correct.\n");
+		LOG_ERR("len: %d, hdr.len: %d\n", len, hdr.len);
+		return NULL;
+	}
+	LOG_DBG("len %u", hdr.len);
+
+	buf = bt_buf_get_evt(hdr.evt, discardable, discardable ? K_NO_WAIT : K_SECONDS(3));
+	if (!buf) {
+		if (discardable) {
+			LOG_DBG("Discardable buffer pool full, ignoring event");
+		} else {
+			LOG_ERR("No available event buffers!");
+
+		}
+		__ASSERT_NO_MSG(buf);
+		return buf;
+	}
+
+	net_buf_add_mem(buf, &hdr, sizeof(hdr));
+
+	buf_tailroom = net_buf_tailroom(buf);
+	if (buf_tailroom < len) {
+		LOG_ERR("Not enough space in buffer %zu/%zu", len, buf_tailroom);
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	net_buf_add_mem(buf, data, len);
+
+	return buf;
+}
+
+static struct net_buf *treat_acl(uint8_t *data, size_t len,
+				 uint8_t *ext_data, uint16_t ext_len)
+{
+	struct bt_hci_acl_hdr hdr;
+	struct net_buf *buf;
+	size_t buf_tailroom;
+
+	if (len < sizeof(hdr)) {
+		LOG_ERR("Not enough data for ACL header");
+		return NULL;
+	}
+
+	buf = bt_buf_get_rx(BT_BUF_ACL_IN, K_NO_WAIT);
+	if (buf) {
+		memcpy((void *)&hdr, data, sizeof(hdr));
+		data += sizeof(hdr);
+		len -= sizeof(hdr);
+	} else {
+		LOG_ERR("No available ACL buffers!");
+		return NULL;
+	}
+
+	if (ext_len != sys_le16_to_cpu(hdr.len)) {
+		LOG_ERR("ACL payload length is not correct");
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	net_buf_add_mem(buf, &hdr, sizeof(hdr));
+	buf_tailroom = net_buf_tailroom(buf);
+	if (buf_tailroom < len) {
+		LOG_ERR("Not enough space in buffer %zu/%zu", len, buf_tailroom);
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	LOG_DBG("ext_len %u", ext_len);
+	net_buf_add_mem(buf, ext_data, ext_len);
+
+	return buf;
+}
+
+static struct net_buf *treat_iso(uint8_t *data, size_t len,
+				 uint8_t *ext_data, uint16_t ext_len)
+{
+	struct bt_hci_iso_hdr hdr;
+	struct net_buf *buf;
+	size_t buf_tailroom;
+
+	if (len < sizeof(hdr)) {
+		LOG_ERR("Not enough data for ISO header");
+		return NULL;
+	}
+
+	buf = bt_buf_get_rx(BT_BUF_ISO_IN, K_NO_WAIT);
+	if (buf) {
+		memcpy((void *)&hdr, data, sizeof(hdr));
+		data += sizeof(hdr);
+		len -= sizeof(hdr);
+	} else {
+		LOG_ERR("No available ISO buffers!");
+		return NULL;
+	}
+
+	if (ext_len != bt_iso_hdr_len(sys_le16_to_cpu(hdr.len))) {
+		LOG_ERR("ISO payload length is not correct");
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	net_buf_add_mem(buf, &hdr, sizeof(hdr));
+	buf_tailroom = net_buf_tailroom(buf);
+	if (buf_tailroom < len) {
+		LOG_ERR("Not enough space in buffer %zu/%zu", len, buf_tailroom);
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	LOG_DBG("ext_len %zu", ext_len);
+	net_buf_add_mem(buf, ext_data, ext_len);
+
+	return buf;
+}
+
+static int receive_data(uint8_t *data, uint16_t len,
+			 uint8_t *ext_data, uint16_t ext_len)
```

Comment:
Why don't the `data` and `ext_data` pointers have `const` here, like you have for `BLECB_Indication()`? It doesn't look like the data needs to be modified, so by adding `const` you could get rid of the explicit casting when you call this function.

Reply:
Indeed. fixed